src/System.Windows.Forms/src/System/Windows/Forms/MonthCalendar.MonthCalendarAccessibleObject.cs

```diff
@@ -214,613 +109,405 @@ internal override int ColumnCount
 
-                switch ((MCHT)hitTestInfo.uHit)
-                {
-                    case MCHT.TITLEBTNPREV:
-                        return GetCalendarChildAccessibleObject(_calendarIndex, CalendarChildType.PreviousButton);
-
-                    case MCHT.TITLEBTNNEXT:
-                        return GetCalendarChildAccessibleObject(_calendarIndex, CalendarChildType.NextButton);
-
-                    case MCHT.TITLE:
-                    case MCHT.TITLEMONTH:
-                    case MCHT.TITLEYEAR:
-                        return GetCalendarChildAccessibleObject(_calendarIndex, CalendarChildType.CalendarHeader);
-
-                    case MCHT.CALENDARDAY:
-                    case MCHT.CALENDARWEEKNUM:
-                    case MCHT.CALENDARDATE:
-                        // Get calendar body's child.
-                        CalendarBodyAccessibleObject? calendarBodyAccessibleObject = GetCalendarChildAccessibleObject(_calendarIndex, CalendarChildType.CalendarBody) as CalendarBodyAccessibleObject;
-                        return calendarBodyAccessibleObject?.GetFromPoint(hitTestInfo);
-
-                    case MCHT.TODAYLINK:
-                        return GetCalendarChildAccessibleObject(_calendarIndex, CalendarChildType.TodayLink);
-                }
-
-                return base.ElementProviderFromPoint(x, y);
-            }
-
-            internal override UiaCore.IRawElementProviderFragment? FragmentNavigate(UiaCore.NavigateDirection direction)
-            {
-                switch (direction)
-                {
-                    case UiaCore.NavigateDirection.FirstChild:
-                        return GetCalendarChildAccessibleObject(_calendarIndex, CalendarChildType.PreviousButton);
-                    case UiaCore.NavigateDirection.LastChild:
-                        return _ownerMonthCalendar.ShowTodayCircle
-                            ? GetCalendarChildAccessibleObject(_calendarIndex, CalendarChildType.TodayLink)
-                            : GetCalendarChildAccessibleObject(_calendarIndex, CalendarChildType.CalendarBody);
-                }
-
-                return base.FragmentNavigate(direction);
+                // See "uHit" kinds in the MS doc:
+                // https://docs.microsoft.com/windows/win32/api/commctrl/ns-commctrl-mchittestinfo
+                return hitTestInfo.uHit switch
+                {
+                    MCHT.CALENDARCONTROL => this,
+                    MCHT.TITLEBTNPREV or MCHT.PREV => PreviousButtonAccessibleObject,
+                    MCHT.TITLEBTNNEXT or MCHT.NEXT => NextButtonAccessibleObject,
+                    MCHT.CALENDARDATEMIN // The given point was over the minimum date in the calendar
+                        => CalendarsAccessibleObjects.First?.Value is CalendarAccessibleObject firstCalendar
+                        && firstCalendar.Bounds.Contains(innerX, innerY)
+                        ? firstCalendar
+                        : this,
```

Comment:
Incorrect indentation
```suggestion
                        => CalendarsAccessibleObjects.First?.Value is CalendarAccessibleObject firstCalendar
                        && firstCalendar.Bounds.Contains(innerX, innerY)
                            ? firstCalendar
                            : this,
```

Ditto below